agent/engine/docker_task_engine.go

```diff
@@ -287,7 +287,9 @@ func (engine *DockerTaskEngine) monitorExecAgentProcesses(ctx context.Context) {
 		}
 		for _, c := range task.Containers {
 			if execcmd.IsExecEnabledContainer(c) {
-				go engine.monitorExecAgentRunning(ctx, mTask, c)
+				if ma, _ := c.GetManagedAgentByName(execcmd.ExecuteCommandAgentName); !ma.InitFailed {
```

Comment:
I'm afraid we need to do the check of `ma.InitFailed` inside of the `engine.monitorExecAgentRunning` function. The reason is that most likely, the agent hasn't been initialized yet when this code is executed. 

After looking at this more closely, it seems that the best approach would be to move the call `go engine.monitorExecAgentRunning(ctx, mTask, c)` after we [initialize the container with no error](https://github.com/aws/amazon-ecs-agent/blob/feature/ecs_exec/agent/engine/docker_task_engine.go#L1145)

Sorry I didn't catch this earlier :(